src/Altinn.App.Core/Internal/App/AppMetadata.cs

```diff
@@ -0,0 +1,118 @@
+                return _application;
+            }
+
+            string filename = _settings.AppBasePath + _settings.ConfigurationFolder + _settings.ApplicationMetadataFileName;
+            try
+            {
+                if (File.Exists(filename))
+                {
+                    JsonSerializerOptions jsonSerializerOptions = new JsonSerializerOptions()
+                    {
+                        PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+                        PropertyNameCaseInsensitive = true,
+                        AllowTrailingCommas = true
+                    };
+                    using FileStream fileStream = File.OpenRead(filename);
+                    var application = await JsonSerializer.DeserializeAsync<ApplicationMetadata>(fileStream, jsonSerializerOptions);
+                    if (application == null)
+                    {
+                        throw new JsonException($"Deserialization returned null, Could indicate problems with deserialization of {filename}");
+                    }
+                    application.App = application.Id.Split("/")[1];
+                    application.Features = await _appFeatures.GetEnabledFeatures();
+                    _application = application;
+
+                    return _application;
+                }
+
+                throw new FileNotFoundException($"Unable to locate application metadata file: {filename}");
+            }
+            catch (JsonException ex)
+            {
+                _logger.LogError(ex, "Something went wrong when parsing application metadata");
+                throw new JsonException("Something went wrong when parsing application metadata", ex);
+            }
+        }
+
+        /// <inheritdoc />
+        public async Task<string?> GetApplicationXACMLPolicy()
+        {
+            string filename = _settings.AppBasePath + _settings.ConfigurationFolder + _settings.AuthorizationFolder + _settings.ApplicationXACMLPolicyFileName;
+            try
+            {
+                if (File.Exists(filename))
+                {
+                    return await File.ReadAllTextAsync(filename, Encoding.UTF8);
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Something went wrong when fetching XACML Policy");
+            }
+
+            return null;
+        }
+
+        /// <inheritdoc />
+        public async Task<string?> GetApplicationBPMNProcess()
+        {
+            string filename = _settings.AppBasePath + _settings.ConfigurationFolder + _settings.ProcessFolder + _settings.ProcessFileName;
+            try
+            {
+                if (File.Exists(filename))
+                {
+                    return await File.ReadAllTextAsync(filename, Encoding.UTF8);
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Something went wrong when fetching BPMNProcess");
```

Comment:
Should we do the same for BPMN as for ApplicationMetadata? The app will crash without it so might as well let it crash and burn.

Reply:
We should do an app validation on startup to check everything as I suggested in https://github.com/Altinn/app-lib-dotnet/pull/98

Reply:
I totally agree we should add some sort of startup validation. If we could make it easy to extend the startup validation we could tackle the files one at a time and also give each app developer the possibility to extend what checks are triggered at startup.

Reply:
Created an issue to track validation on startup #208 to not make this PR even larger